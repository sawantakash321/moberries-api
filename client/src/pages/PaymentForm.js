import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';
import React from 'react';

const PaymentForm = props => {
  const {
    values,
    handleChange,
    errors: {errors},
  } = props;

  return (
    <React.Fragment>
      <Typography variant="h6" gutterBottom>
        Payment method
      </Typography>
      <Grid container spacing={24}>
        <Grid item xs={12}>
          <TextField
            type="number"
            id="creditNumber"
            label="Credit Card Number"
            value={values.creditNumber}
            onChange={handleChange('creditNumber')}
            name="creditNumber"
            margin="normal"
            fullWidth
            placeholder="Enter 16 digit card number"
            variant="outlined"
            error={values.creditNumber === ''}
            helperText={errors.errors.creditNumber}
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            type="number"
            id="expDate"
            label="Credit Card Expire Date"
            value={values.expDate}
            onChange={handleChange('expDate')}
            placeholder="Enter date in MMYYYY format"
            name="expDate"
            margin="normal"
            variant="outlined"
            error={values.expDate === ''}
            helperText={errors.errors.expDate}
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            type="number"
            id="cvv"
            label="CVV"
            value={values.cvv}
            onChange={handleChange('cvv')}
            placeholder="Enter 3 digit CVV number "
            name="cvv"
            margin="normal"
            variant="outlined"
            error={values.cvv === ''}
            helperText={errors.errors.cvv}
          />
        </Grid>
        <Grid item xs={12}>
          <FormControlLabel
            control={<Checkbox color="secondary" name="saveCard" value="yes" />}
            label="Remember credit card details for next time"
          />
        </Grid>
      </Grid>
    </React.Fragment>
  );
};

PaymentForm.propTypes = {
  values: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default PaymentForm;
