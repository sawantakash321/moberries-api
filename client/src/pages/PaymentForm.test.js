'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import PaymentForm from './PaymentForm';

// Snapshot test
it('matches the snapshot', () => {
  const values = {
    creditNumber: 32123547361236,
    cvv: 234,
    expDate: 234223,
  };
  const errors = {
    errors: {
      creditNumber: '',
      cvv: '',
      expDate: '',
    },
  };
  const handleChange = jest.fn();
  const tree = renderer.create(<PaymentForm values={values} errors={{errors}} handleChange={handleChange} />).toJSON();
  expect(tree).toMatchSnapshot();
});
