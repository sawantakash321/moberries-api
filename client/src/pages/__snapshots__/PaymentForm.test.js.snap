// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
Array [
  <h6
    className="MuiTypography-root-1 MuiTypography-h6-18 MuiTypography-gutterBottom-28"
  >
    Payment method
  </h6>,
  <div
    className="MuiGrid-container-37 MuiGrid-spacing-xs-24-61"
  >
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77"
    >
      <div
        className="MuiFormControl-root-134 MuiFormControl-marginNormal-135 MuiFormControl-fullWidth-137"
      >
        <label
          className="MuiFormLabel-root-149 MuiFormLabel-filled-153 MuiInputLabel-root-138 MuiInputLabel-formControl-143 MuiInputLabel-animated-146 MuiInputLabel-shrink-145 MuiInputLabel-outlined-148"
          data-shrink={true}
          htmlFor="creditNumber"
        >
          Credit Card Number
        </label>
        <div
          className="MuiInputBase-root-169 MuiOutlinedInput-root-156 MuiInputBase-fullWidth-178 MuiInputBase-formControl-170"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-186 MuiOutlinedInput-notchedOutline-163"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-187"
              style={
                Object {
                  "width": 0,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <input
            aria-invalid={false}
            className="MuiInputBase-input-179 MuiOutlinedInput-input-164 MuiInputBase-inputType-182"
            disabled={false}
            id="creditNumber"
            name="creditNumber"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter 16 digit card number"
            required={false}
            type="number"
            value={32123547361236}
          />
        </div>
        
      </div>
    </div>
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77 MuiGrid-grid-md-6-99"
    >
      <div
        className="MuiFormControl-root-134 MuiFormControl-marginNormal-135"
      >
        <label
          className="MuiFormLabel-root-149 MuiFormLabel-filled-153 MuiInputLabel-root-138 MuiInputLabel-formControl-143 MuiInputLabel-animated-146 MuiInputLabel-shrink-145 MuiInputLabel-outlined-148"
          data-shrink={true}
          htmlFor="expDate"
        >
          Credit Card Expire Date
        </label>
        <div
          className="MuiInputBase-root-169 MuiOutlinedInput-root-156 MuiInputBase-formControl-170"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-186 MuiOutlinedInput-notchedOutline-163"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-187"
              style={
                Object {
                  "width": 0,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <input
            aria-invalid={false}
            className="MuiInputBase-input-179 MuiOutlinedInput-input-164 MuiInputBase-inputType-182"
            disabled={false}
            id="expDate"
            name="expDate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter date in MMYYYY format"
            required={false}
            type="number"
            value={234223}
          />
        </div>
        
      </div>
    </div>
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77 MuiGrid-grid-md-6-99"
    >
      <div
        className="MuiFormControl-root-134 MuiFormControl-marginNormal-135"
      >
        <label
          className="MuiFormLabel-root-149 MuiFormLabel-filled-153 MuiInputLabel-root-138 MuiInputLabel-formControl-143 MuiInputLabel-animated-146 MuiInputLabel-shrink-145 MuiInputLabel-outlined-148"
          data-shrink={true}
          htmlFor="cvv"
        >
          CVV
        </label>
        <div
          className="MuiInputBase-root-169 MuiOutlinedInput-root-156 MuiInputBase-formControl-170"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-186 MuiOutlinedInput-notchedOutline-163"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-187"
              style={
                Object {
                  "width": 0,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <input
            aria-invalid={false}
            className="MuiInputBase-input-179 MuiOutlinedInput-input-164 MuiInputBase-inputType-182"
            disabled={false}
            id="cvv"
            name="cvv"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter 3 digit CVV number "
            required={false}
            type="number"
            value={234}
          />
        </div>
        
      </div>
    </div>
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77"
    >
      <label
        className="MuiFormControlLabel-root-188"
      >
        <span
          className="MuiButtonBase-root-210 MuiIconButton-root-204 MuiPrivateSwitchBase-root-200 MuiCheckbox-root-194 MuiCheckbox-colorSecondary-199"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-209"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-213"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              className="MuiPrivateSwitchBase-input-203"
              data-indeterminate={false}
              name="saveCard"
              onChange={[Function]}
              type="checkbox"
              value="yes"
            />
          </span>
          <span
            className="MuiTouchRipple-root-222"
          />
        </span>
        <span
          className="MuiTypography-root-1 MuiTypography-body1-10 MuiFormControlLabel-label-193"
        >
          Remember credit card details for next time
        </span>
      </label>
    </div>
  </div>,
]
`;
